src/tally.cpp

```diff
@@ -420,34 +448,103 @@ std::string pyne::Tally::mcnp(int tally_index, std::string mcnp_version) {
   output << "C " << tally_name << std::endl;
   output << std::setiosflags(std::ios::fixed) << std::setprecision(6);
 
-  if (normalization > 1.0)
+  if (normalization > 1.0) 
     output << std::scientific;
+  int tally_id = 0;
 
   // neednt check entity type
   if (entity_type.find("Surface") != std::string::npos) {
     if (tally_type.find("Current") != std::string::npos) {
-      output << form_mcnp_tally(tally_index, 1, particle_token, 
-                                entity_id, entity_size, normalization);
+      tally_id = 1;
     } else if (tally_type.find("Flux") != std::string::npos) {
-      output << form_mcnp_tally(tally_index, 2, particle_token, 
-                                entity_id, entity_size, normalization);
+      tally_id = 2;
     }
+    output << form_mcnp_tally(tally_index, tally_id, particle_token, entity_id,
+                              entity_size, normalization);
 
   } else if (entity_type.find("Volume") != std::string::npos) {
     if (tally_type.find("Flux") != std::string::npos) {
-      output << form_mcnp_tally(tally_index, 4, particle_token, 
-                                entity_id, entity_size, normalization);
+      tally_id = 4;
     } else if (tally_type.find("Current") != std::string::npos) {
       // makes no sense in mcnp
+      return "";
+    }
+    output << form_mcnp_tally(tally_index, tally_id, particle_token, entity_id,
+                              entity_size, normalization);
+  } else if (entity_type.find("Mesh") != std::string::npos) {
+    output << "FMESH" << tally_index << "4:" << particle_token << "  ";
```

Comment:
What if this was in a method called `form_mcnp_mesh_tally()` to be parallel with the normal tallies?